packages/core/src/state-management/state-manager.ts

```diff
@@ -200,16 +205,25 @@ export class StateManager {
    *
    * @param streamId
    * @param tip - Stream Tip CID
-   * @private
+   * @param model - Model Stream ID
    */
-  handlePubsubUpdate(streamId: StreamID, tip: CID): void {
-    this.fromMemoryOrStore(streamId).then((state$) => {
+  async handlePubsubUpdate(streamId: StreamID, tip: CID, model?: StreamID): Promise<void> {
+    if (model && this._index.shouldIndexStream(model)) {
+      const inPinStore = await this.pinStore.ls(streamId)
```

Comment:
this results in going to the state store twice, since it also happens below in `fromMemoryOrStore`.  You should be able to combine the logic

Reply:
I think you should be able to do something like:
```
let state$ = await this.fromMemoryOrStore(streamId)
const shouldIndex = model && this._index.shouldIndexStream(model)
if (!shouldIndex && !state$) {
  // stream isn't pinned or indexed, nothing to do
  return
}

if (!state$) {
  state$ = await this.load(streamId)
}
this.executionQ.forStream(streamId).add(async() => {
  await this._handleTip(state$, tip)
}
this.indexStreamIfNeeded(state$)
```

Reply:
Oh, I see what you mean and your example should work to address it the initial issue of it not being found. If possible, I'd love to have a follow up chat about this though to dive a bit deeper into the execution queue. I was under the impression that we still have to add it first to the state store for it then to be properly processed.